app/code/Magento/CatalogGraphQl/Model/Resolver/Product/PriceRange.php

```diff
@@ -130,4 +133,39 @@ private function formatPrice(float $regularPrice, float $finalPrice, StoreInterf
             'discount' => $this->discount->getDiscountByDifference($regularPrice, $finalPrice),
         ];
     }
+
+    /**
+     * Check if the product is allowed to show price
+     *
+     * @param ProductInterface $product
+     * @return bool
+     */
+    private function canShowPrice($product): bool
+    {
+        if ($product->hasData('can_show_price') && $product->getData('can_show_price') === false) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Format empty result
+     *
+     * @return array
+     */
+    private function formatEmptyResult(): array
+    {
+        return [
```

Comment:
If we could make the Money! type not mandatory, would be better.
We won't now though, because it's backward incompatible in schema.  I guess we never thought a price can be optional...

Reply:
Yes, so this seemed like the good approach under these circumstances.